tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -616,7 +616,9 @@ class ScopedPoolingDescriptor {
       LOG(FATAL) << "could not create cudnn pooling descriptor: "
                  << ToString(status);
     }
-
+    bool propagate_nans;
+    tensorflow::ReadBoolFromEnvVar("TF_ENABLE_MAXPOOL_NANPROP",
```

Comment:
If I remember correct, this function is called very time the op is called. Can we cache the value on the first reading? There are a number of examples like that in this file.

Reply:
Most other examples of caching envars are static, but we need to be able to toggle the value in order to test both propagating and non-propagating nans conveniently. This will require plumbing the propagation flag through from the MaxPoolingGradOp<> class in maxpooling_op.cc. I'll also also add a nan-prop field to the dnn::PoolingDescriptor wrapper class.

Reply:
In that case, how about having the caller should pass it in. TensorFlow is better equipped to cache the value.

Reply:
How is this?